### ISSUE TITLE
API feature request : Post hiding and other post-related API actions should return the full post response.

### ISSUE THREAD

**Issue by dessalines** [LemmyNet/lemmy#4755]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?

### Is your proposal related to a problem?

Currently in the code, we have several post actions returning a `SuccessResponse`.  While this works, its not ideal for front ends, who often could use the post response, to update fields in their UI. It can use abstracted `updatePost` methods to do this, rather than relying on individual logic per each success response.

### Describe the solution you'd like.

IE a `DeletePost` updates the `post.deleted` field, and `HidePost` should work the same way.

### Describe alternatives you've considered.

NA

### Additional context

NA

**Comment by dessalines** [comment]
See above: 

> Currently in the code, we have several post actions returning a SuccessResponse. While this works, its not ideal for front ends, who often could use the post response, to update fields in their UI. It can use abstracted updatePost methods to do this, rather than relying on individual logic per each success response.

This is a specific request from 2 other app devs, and one that I need also in jerboa(returning the full post after doing an action on it.)

I also have an open PR for bulk mark read already: #5178  , which returns the successresponse , so no functionality is actually being removed.

Also please read our [code of conduct](https://join-lemmy.org/docs/code_of_conduct.html); I personally don't like getting downvotes on issues and PRs for fixing problems, as I'm sure you wouldn't either.

**Comment by aeharding** [comment]
> I also have an open PR for bulk mark read already: https://github.com/LemmyNet/lemmy/pull/5178 , which returns the successresponse , so no functionality is actually being removed.

This is true, however it introduces a breaking change which all apps that mark as read will have to support. Also now maintain two endpoints. This seems like a high burden to me, especially given how easy this is to handle by each app. That is why I am opposed to the premise of this PR.

> Also please read our [code of conduct](https://join-lemmy.org/docs/code_of_conduct.html); I personally don't like getting downvotes on issues and PRs for fixing problems, as I'm sure you wouldn't either.

My apologies, it was not my intention to be rude. I use the thumbs down reaction to express disapproval with the premise of issues and/or PRs. But I will remove it :)

**Comment by dessalines** [comment]
Its no probs.

Its okay that two endpoints are needed, they have different explicit return types, so its best to keep them separate. Its a case of several apps needing different functionality, so we gotta be flexible in supporting everyone's needs.
